### PR TITLE
Blog App - processing data in models 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# Blog App - processing data in models 🎉 🚀 ✔️ 
- Set up associations between models.
- Write SQL queries with Active Record.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'user_id'
+  has_many :comments, foreign_key: 'user_id'
+  has_many :likes, foreign_key: 'user_id'
 end
```
Comment (thecodechaser) [inline app/models/user.rb]: - [ ] Please make sure to add the below methods in the User model as requested in the project requirements.
  - A method that returns the 3 most recent posts for a given user.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes
 end
```
Comment (thecodechaser) [inline app/models/post.rb]: - [ ] Please make sure to add the below methods in the Post model as requested in the project requirements.
   - A method that updates the posts counter for a user.
   - A method that returns the 5 most recent comments for a given post.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
 end
```
Comment (thecodechaser) [inline app/models/comment.rb]: - [ ] Please make sure to add the below methods in the Comment model as requested in the project requirements.
  - A method that updates the comments counter for a post.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,2 +1,4 @@
 class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
 end
```
Comment (thecodechaser) [inline app/models/like.rb]: - [ ] Please make sure to add the below methods in the Like model as requested in the project requirements.
  - A method that updates the likes counter for a post.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Nizamuddin4493👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Associations are set up correctly. ✔️
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Gitflow is followed. :heavy_check_mark:



## Required Changes ♻️

- [ ] The License link in the README file is broken, Please make sure to fix it to make your README file more professional.
- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @Nizamuddin4493  

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**